obspy/io/gse2/bulletin.py

```diff
@@ -0,0 +1,1117 @@
+        # errors are set.
+        for i in range(3):
+            magnitude = magnitudes[i]
+            if magnitude is not None:
+                public_id = "magnitude/%s" % i
+                mag_res_id = \
+                    self._get_res_id(public_id, parent_res_id=origin_res_id)
+                magnitude.resource_id = mag_res_id
+
+                if event.preferred_magnitude_id is None:
+                    event.preferred_magnitude_id = magnitude.resource_id.id
+
+        return origin, origin_res_id
+
+    def _parse_first_line_origin(self, line, event, magnitudes):
+        """
+        Parse the first line of origin data.
+
+        :type line: str
+        :param line: Line to parse.
+        :type event: :class:`~obspy.core.event.Event`
+        :param event: Event of the origin.
+        :type magnitudes: list of :class:`~obspy.core.event.Magnitude`
+        :param magnitudes: Store magnitudes in a list to keep
+            their positions.
+        :rtype: :class:`~obspy.core.event.origin.Origin`,
+            :class:`~obspy.core.event.ResourceIdentifier`
+        :returns: Parsed origin or None, resource identifier of the
+            origin.
+        """
+        magnitude_types = []
+        magnitude_values = []
+        magnitude_station_counts = []
+
+        fields = self.fields['line_1']
+
+        time_origin = line[fields['time']].strip()
+        time_fixed_flag = line[fields['time_fixf']].strip()
+        latitude = line[fields['lat']].strip()
+        longitude = line[fields['lon']].strip()
+        epicenter_fixed_flag = line[fields['epicenter_fixf']].strip()
+        depth = line[fields['depth']].strip()
+        depth_fixed_flag = line[fields['depth_fixf']].strip()
+        phase_count = line[fields['n_def']].strip()
+        station_count = line[fields['n_sta']].strip()
+        azimuthal_gap = line[fields['gap']].strip()
+        magnitude_types.append(line[fields['mag_type_1']].strip())
+        magnitude_values.append(line[fields['mag_1']].strip())
+        magnitude_station_counts.append(line[fields['mag_n_sta_1']].strip())
+        magnitude_types.append(line[fields['mag_type_2']].strip())
+        magnitude_values.append(line[fields['mag_2']].strip())
+        magnitude_station_counts.append(line[fields['mag_n_sta_2']].strip())
+        magnitude_types.append(line[fields['mag_type_3']].strip())
+        magnitude_values.append(line[fields['mag_3']].strip())
+        magnitude_station_counts.append(line[fields['mag_n_sta_3']].strip())
+        author = line[fields['author']].strip()
+        origin_id = line[fields['id']].strip()
+
+        origin = Origin()
+        origin.quality = OriginQuality()
+
+        try:
+            origin.time = UTCDateTime(time_origin.replace('/', '-'))
+            origin.latitude = float(latitude)
+            origin.longitude = float(longitude)
+        except (TypeError, ValueError):
+            self._warn('Missing origin data, skipping event')
```

Comment:
Line number is actually printed. I created a custom `_warn` method for that. I added a description for this method since it wasn't clear enough. I should maybe renamed it.